pkg/utils/gardener/seed.go

```diff
@@ -92,21 +92,41 @@ func hasExactUsages(usages, requiredUsages []certificatesv1.KeyUsage) bool {
 	return true
 }
 
-// GetWildcardCertificate gets the wildcard certificate for the ingress domain.
+// GetWildcardCertificate gets the wildcard TLS certificate for the seed ingress domain.
 // Nil is returned if no wildcard certificate is configured.
 func GetWildcardCertificate(ctx context.Context, c client.Client) (*corev1.Secret, error) {
+	return getWildcardCertificate(ctx, c, v1beta1constants.GardenRoleControlPlaneWildcardCert)
+}
+
+// GetGardenWildcardCertificate gets the wildcard TLS certificate for the Garden runtime ingress domain.
+// Nil is returned if no wildcard certificate is configured.
+func GetGardenWildcardCertificate(ctx context.Context, c client.Client) (*corev1.Secret, error) {
+	secret, error := getWildcardCertificate(ctx, c, v1beta1constants.GardenRoleGardenWildcardCert)
+	if error != nil {
+		return nil, error
+	}
+	if secret == nil {
+		// try to lookup secret with old role name
+		secret, error = getWildcardCertificate(ctx, c, v1beta1constants.GardenRoleControlPlaneWildcardCert)
```

Comment:
Please add TODO to remove this in the next release. It's undocumented and unwanted side effects may occur, .e.g., if only wildcard cert for seed is configured. Please also announce deprecation in release note.